### PR TITLE
add ci for vsphere multi-vcenter

### PR DESCRIPTION
**Pull request by WenXinWei**
This pr is aid to test and add ci for vsphere multi-vcenter(IPI), epic :SPLAT-1111.
and also fix failure job:
nightly-vsphere-ipi-proxy-tp-f28
nightly-vsphere-ipi-disktype-xxx

### CODE REVIEW

**Comment by WenXinWei** [comment]
pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-multi-vcenter-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-multi-vcenter-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-multi-vcenter-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-disktype-thick-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-debug-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-debug-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-debug-vsphere

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-debug-vsphere

**Comment by WenXinWei** [comment]
job:nightly-vsphere-ipi-proxy-tp-f28 failed because command `"oc get cm cloud-provider-config -n openshift-config -o yaml"` can not get resource pool path when enable TP. so I remove the step in chain:cucushift-installer-rehearse-vsphere-ipi-proxy-provision(the failure TP job use this chain with env : FEATURE_SET: TechPreviewNoUpgrade ).  ref: check-resourcepool covered in job:vsphere-ipi-f28-longduration-hive.  @jianlinliu @jinyunma  please help review when you have time.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml [hunk 1]
```diff
@@ -3015,6 +3015,18 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-multi-vcenter-f28
+  cron: 4 4 12 * *
+  steps:
+    cluster_profile: vsphere-elastic
+    env:
+      FEATURE_GATES: '["ClusterAPIInstall=true","VSphereMultiVCenters=true"]'
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3023]: Sounds like `ClusterAPIInstall=true` is to enable capi, by default, vsphere is running installation with capi in 4.17 ?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml [hunk 2]
```diff
@@ -3015,6 +3015,18 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-multi-vcenter-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3018]: We also need a destructive ci job.

**Comment by jianlinliu** [comment]
> job:nightly-vsphere-ipi-proxy-tp-f28 failed because command `"oc get cm cloud-provider-config -n openshift-config -o yaml"` can not get resource pool path when enable TP. 

In a normal install, it works well, only happens in TP jobs ? If yes, I prefer to drop the step of `oc get cm cloud-provider-config`, because we already checked vm via govc commands in the prior check points.

> so I remove the step in chain:cucushift-installer-rehearse-vsphere-ipi-proxy-provision(the failure TP job use this chain with env : FEATURE_SET: TechPreviewNoUpgrade ). ref: check-resourcepool covered in job:vsphere-ipi-f28-longduration-hive.

It is better not to rely on other team's dedicated ci jobs to check installer's test cases, so I suggest to keep it in the proxy install workflow by fixing the above issues.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3018]: ok. i will add it.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml (hunk 1 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3023]: yes, i copy this config from e2e job. do you mean we should remove this feature gate setting?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3023]: If capi is already the default behavior on 4.17 vsphere, then we do not need to set it here, only keep `VSphereMultiVCenters=true`

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-multi-vcenter-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-proxy-tp-f28

**Comment by jianlinliu** [comment]
/lgtm

@WenXinWei do you have any update or check on this PR ?

**Comment by WenXinWei** [comment]
> /lgtm
> 
> @WenXinWei do you have any update or check on this PR ?

nothing more

**Comment by jianlinliu** [comment]
cc @liangxia to review.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml (hunk 2 above)
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3018]: Do we need to test it in 4.18 ?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3018]: Good point. @WenXinWei pls also add them to 4.18.

**Comment by WenXinWei** [comment]
job added in 4.18  @liangxia @jianlinliu

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3018]: @liangxia it is done.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml [hunk 3]
```diff
@@ -3015,6 +3015,30 @@ tests:
...
+  cron: 4 4 12 * *
+  steps:
+    cluster_profile: vsphere-elastic
+    env:
+      FEATURE_GATES: '["VSphereMultiVCenters=true"]'
+      FEATURE_SET: CustomNoUpgrade
+      POOLS: vcenter.ci.ibmc.devcluster.openshift.com-cidatacenter-cicluster vcenter.ci.ibmc.devcluster.openshift.com-cidatacenter-1-cicluster-1
+        vcenter-1.ci.ibmc.devcluster.openshift.com-cidatacenter-2-cicluster-3
```
[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:3026]: It is a good idea, @WenXinWei can you update it ?

**Pull request merged**